zerver/lib/mention.py

```diff
@@ -28,6 +28,6 @@ def extract_name(s):
 def possible_mentions(content):
     # type: (Text) -> Set[Text]
     matches = re.findall(find_mentions, content)
-    names = {extract_name(match) for match in matches}
-    names = {name for name in names if name}
+    names_with_none = {extract_name(match) for match in matches}
```

Comment:
This code bugged me in that it makes a whole `set` before throwing it away to make nearly the same `set`. Sadly Python's comprehensions lack a way to write these cleanly as one comprehension, but I ended up adding a tiny tweak here to make `names_with_none` a genexp.